tests/phpunit/tests/block-template.php

```diff
@@ -275,6 +275,82 @@ public function test_get_the_block_template_html_does_not_generally_enforce_loop
 		$this->assertSame( array( false, true ), $in_the_loop_logs, 'Main query loop was triggered incorrectly' );
 	}
 
+	/**
+	 * @ticket 58319
+	 *
+	 * @covers ::get_block_theme_folders
+	 */
+	public function test_get_block_theme_folders_block_theme() {
+		$expected = array(
+			'wp_template'      => 'templates',
+			'wp_template_part' => 'parts',
+		);
+		$this->assertSame( $expected, get_block_theme_folders( 'block-theme' ), 'Expected `templates` and `parts` and `wp_template` and `wp_template_part` respectively.' );
```

Comment:
Please add a test for a child theme.

Reply:
I have added this to the new dataProvider which covers few more cases.